lib/neovim-plugin.nix

```diff
@@ -161,8 +162,43 @@
                   # Add the plugin setup code `require('foo').setup(...)` to the lua configuration
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
 
-                  # Write the lua configuration `luaConfig.content` to the config file
-                  (setLuaConfig cfg.luaConfig.content)
+                  # Write the lua configuration `luaConfig.content` to the config file when lazy loading is not enabled
+                  (lib.mkIf (!cfg.lazyLoad.enable) (setLuaConfig cfg.luaConfig.content))
+
+                  # When lazy loading is enabled for this plugin, route its configuration to the enabled provider
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                      {
+                        assertion = cfg.lazyLoad.enable && (config.plugins.lz-n.enable || config.plugins.lazy.enable);
```

Comment:
For a future PR it could be usefull to have a way to enable lazyLoading without a provider, in order to implement lazy  loading for a  provider not yet supported by nixvim

Reply:
Thinking this should just be a warning instead of an assertion, then?

Reply:
I think the reason it's an assertion right now, is because if you use this option we handle the plumbing and we can only guarantee it works for things we have wired up.

Reply:
I think assertion is best for now, but we could address ways to bypass the assertion in a future PR?

Reply:
Hmm.. I just switched to a warning just because I saw the potential for enabling lazy loading just to pass the luaConfig around yourself. ie: Someone wants to try using it with lazy before we officially support it.

Reply:
Yeah my first thoughts was doing it in a subsequent PR, but no problem if it is being done here.